src/main/java/mate/academy/dao/MovieDaoImpl.java

```diff
@@ -0,0 +1,49 @@
+package mate.academy.dao;
+
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Movie;
+import mate.academy.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import java.util.Optional;
+
+@Dao
+public class MovieDaoImpl implements MovieDao {
+    @Override
+    public Movie save(Movie movie) {
+        SessionFactory sessionFactory = HibernateUtil.getSessionFactory();
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.save(movie);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw  new DataProcessingException("Cant save movie "
+                    + movie.getTitle() + " to DB", e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+        return movie;
+    }
+
+    @Override
+    public Optional<Movie> get(Long id) {
+        SessionFactory sessionFactory = HibernateUtil.getSessionFactory();
+        try (Session session = sessionFactory.openSession()) {
+            Movie movie;
+            movie = session.get(Movie.class, id);
```

Comment:
```suggestion
            Movie movie = session.get(Movie.class, id);
```
 we don't need to split this statement